FatmaQunnies/Utils.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+def validate_date(date_str):
+    while True:
```

Comment:
The validation function should focus only on the validation, so user input shouldn't be part of the validation process. Instead, we should store all the user input data as-is (without any validation) in a dictionary with key-value pairs. Then, you can either iterate over that dictionary to run the validation functions OR create a class—let's say UserTaskInputData and perform all the validations within that class.